packages/pull-signal/src/services/interop.service.ts

```diff
@@ -4,33 +4,35 @@ import { interopRepository } from "../repositories/interop.repository.js";
 
 interface IInteropService {
   readonly consumerIsAuthorizedToPullSignals: (
-    purposeId: string,
+    consumerId: string,
     eserviceId: string,
     logger: Logger
   ) => Promise<void>;
 }
 export function interopServiceBuilder(db: DB): IInteropService {
   return {
     async consumerIsAuthorizedToPullSignals(
-      purposeId: string,
+      consumerId: string,
       eserviceId: string,
       logger: Logger
     ): Promise<void> {
       logger.info(
-        `InteropService::consumerIsAuthorizedToPullSignals with purposeId: ${purposeId}`
+        `InteropService::consumerIsAuthorizedToPullSignals with consumerId: ${consumerId}`
       );
-      const purposeState = "ACTIVE";
+      const eserviceState = "PUBLISHED";
       const agreementState = "ACTIVE";
+      const purposeState = "ACTIVE";
       const result = await interopRepository(db).findBy(
         eserviceId,
-        purposeId,
+        consumerId,
+        eserviceState,
         purposeState,
         agreementState
       );
-      if (result === null) {
+      if (!result?.length) {
```

Comment:
should be move this into a method? just to read and understand what code do here

Reply:
Got it! This Is a code with a "missed abstraction level".